pkg/stanza/fileconsumer/config.go

```diff
@@ -62,61 +63,50 @@ type Config struct {
 
 // Build will build a file input operator from the supplied configuration
 func (c Config) Build(logger *zap.SugaredLogger, emit EmitFunc) (*Manager, error) {
-	if emit == nil {
-		return nil, fmt.Errorf("must provide emit function")
+	if err := c.validate(); err != nil {
+		return nil, err
 	}
 
-	if len(c.Include) == 0 {
-		return nil, fmt.Errorf("required argument `include` is empty")
+	// Ensure that splitter is buildable
+	factory := newMultilineSplitterFactory(c.Splitter.EncodingConfig, c.Splitter.Flusher, c.Splitter.Multiline)
+	if _, err := factory.Build(int(c.MaxLogSize)); err != nil {
+		return nil, err
 	}
 
-	// Ensure includes can be parsed as globs
-	for _, include := range c.Include {
-		_, err := doublestar.PathMatch(include, "matchstring")
-		if err != nil {
-			return nil, fmt.Errorf("parse include glob: %w", err)
-		}
-	}
+	return c.buildManager(logger, emit, factory)
+}
 
-	// Ensure excludes can be parsed as globs
-	for _, exclude := range c.Exclude {
-		_, err := doublestar.PathMatch(exclude, "matchstring")
-		if err != nil {
-			return nil, fmt.Errorf("parse exclude glob: %w", err)
-		}
+// BuildWithSplitFunc will build a file input operator with customized splitFunc function
+func (c Config) BuildWithSplitFunc(
+	logger *zap.SugaredLogger, emit EmitFunc, splitFunc bufio.SplitFunc) (*Manager, error) {
+	if err := c.validate(); err != nil {
+		return nil, err
 	}
 
-	if c.MaxLogSize <= 0 {
-		return nil, fmt.Errorf("`max_log_size` must be positive")
+	if splitFunc == nil {
+		return nil, fmt.Errorf("must provide split function")
 	}
 
-	if c.MaxConcurrentFiles <= 1 {
-		return nil, fmt.Errorf("`max_concurrent_files` must be greater than 1")
+	// Ensure that splitter is buildable
+	factory := newCustomizeSplitterFactory(c.Splitter.Flusher, splitFunc)
+	if _, err := factory.Build(int(c.MaxLogSize)); err != nil {
+		return nil, err
 	}
 
-	if c.FingerprintSize == 0 {
-		c.FingerprintSize = DefaultFingerprintSize
-	} else if c.FingerprintSize < MinFingerprintSize {
-		return nil, fmt.Errorf("`fingerprint_size` must be at least %d bytes", MinFingerprintSize)
-	}
+	return c.buildManager(logger, emit, factory)
+}
 
-	// Ensure that splitter is buildable
-	factory := newMultilineSplitterFactory(c.Splitter.EncodingConfig, c.Splitter.Flusher, c.Splitter.Multiline)
-	_, err := factory.Build(int(c.MaxLogSize))
+func (c Config) buildManager(logger *zap.SugaredLogger, emit EmitFunc, factory splitterFactory) (*Manager, error) {
+	if emit == nil {
+		return nil, fmt.Errorf("must provide emit function")
+	}
+	startAtBeginning, err := judgeStartAtBeginning(c.StartAt)
 	if err != nil {
 		return nil, err
 	}
-
-	var startAtBeginning bool
-	switch c.StartAt {
-	case "beginning":
-		startAtBeginning = true
-	case "end":
-		startAtBeginning = false
-	default:
-		return nil, fmt.Errorf("invalid start_at location '%s'", c.StartAt)
+	if c.FingerprintSize == 0 {
+		c.FingerprintSize = DefaultFingerprintSize
```

Comment:
I think we can remove this. It is handled during unmarshaling of config.

Importantly, this PR changed the order in which we check this value. It used to check if the value was `0` first (representing that the user did not specify a value, so we should use the default). Then it would check if it was otherwise too small. Now it checks if too small first, so this case will never be true anyways - so now we have two reasons to remove it :)

Reply:
I will remove it